lib/src/async/stream_buffer.dart

```diff
@@ -149,7 +149,7 @@ class StreamBuffer<T> implements StreamConsumer {
     Completer streamDone = new Completer();
     _sub = stream.listen((items) {
       _chunks.add(items);
-      _counter += items is List ? items.length : 1;
+      _counter += items is List ? (items as List).length : 1;
```

Comment:
I think not, if I read the code correctly, because of the Dart rules about type promotion.  In regular Dart, items is dynamic, and promotion is not allowed to promote from dynamic.  We allow promotion from dynamic in strong mode, but we still don't allow promotion between non-subtype related types, and I believe (if I read this correctly) that items will be inferred to have type T, and T and List are not subtype related, and so no type promotion will be done.